docs/DeveloperGuide.md

```diff
@@ -617,6 +617,28 @@ person on the list is born before 12-12-2020.
 
 ### Adding a friend group: `add-group`
 
+Prerequisites: List all person using the list command. There is at least a person present in the list. 
+There's only one existing group with the name `Colleagues` that does not have person with index 3 inside it.
+
+1. Adding a non-existing group
+    
+    1. Test case: `add-group n/Close Friends p/1`<br>
+       Expected: A success message is shown in the status message. `Close Friends` will be selected
+       with the person who was previously index `1` being the only one shown in the Friend Panel.
+   
+    2. Test case: `add-group n/Close Friends p/0`(Invalid index)<br>
+       Expected: Group is not added. Error details will be shown in the status message. 
+       
+2. Adding an existing group
+    
+    1. Test case: `add-group n/Colleagues p/3`<br>
+       Expected: A success message is shown in the status message. `Colleagues` will be selected with, showing all
+       three friends shown in the Friend Panel.
+   
+    2. Test case: `add-group n/Colleagues p/2`<br>
```

Comment:
I think u need to emphasize here that your test case needs to be run from a certain start state because after adding the new person into the group, I should have 3 people in `Colleagues` instead of 2.